docs/under-the-hood/storage.md

```diff
@@ -1,138 +1,201 @@
+$
+\begin{aligned}
+\texttt{StorageRAMUsage} &= 21,723 \times 260\text{B} + 682,943 \times 180\text{B} \\ &= 5,647,980\text{B} + 122,929,740\text{B}\\ &= 128,577,720\text{B} \approx 125\text{MB}
+\end{aligned}
+$
 
-Behavior of these mechanisms can be tweaked in the configuration file,
-usually found in `/etc/memgraph/memgraph.conf`.
+Now, let's run an empty Memgraph instance on a x86 Ubuntu. It consumes **~75MB**
+of RAM due to baseline runtime overhead. Once the dataset is loaded, RAM usage
+rises up to **~260MB**. Memory usage primarily consists of storage and query
+execution memory usage. After executing `FREE MEMORY` query to force the cleanup
+of query execution, the RAM usage drops to **~200MB**. If the baseline runtime
+overhead of **75MB** is subtracted from the total memory usage of the dataset,
+which is **200MB**, and storage memory usage comes up to **~125MB**, which shows
+that the formula is correct.
 
-Check the reference guide on Configuration to see the [possible configuration
-settings for storage](../reference-guide/configuration#storage).
+## The calculation in detail
 
-:::caution
-Snapshot and WAL files are (currently) not compatible between Memgraph
-versions.
+Let's dive deeper into the memory usage values. Because Memgraph works on the
+x86 architecture, calculations are based on the x86 Linux memory usage.
+
+:::tip 
+For the latest and most precise memory layout please clone
+[Memgraph](https://github.com/memgraph/memgraph) and use, e.g.,
+[pahole](https://github.com/PhilArmstrong/pahole-gdb) to discover accurate
+information. 
 :::
 
-## Storable data types
+Each `Vertex` and `Edge` object has a pointer to a `Delta` object. The
+`Delta` object stores all changes on a certain `Vertex` or `Edge` and that's
+why `Vertex` and `Edge` memory usage will be increased by the memory of
+the `Delta` objects they are pointing to. If there are few updates, there are
+also few `Delta` objects because the latest data is stored in the object.
+But, if the database has a lot of concurrent operations, many `Delta` objects
+will be created. Of course, the `Delta` objects will be kept in memory as long as
+needed, and a bit more, because of the internal GC inefficiencies.
+
+### Delta memory layout
+
+Each `Delta` object has a least **104B**.
+
+### Vertex memory layout
+
+Each `Vertex` object has at least **112B** + **104B** for the `Delta` object, in
+total, a minimum of **216B**.
+
+### Edge memory layout
+
+Each `Edge` object has at least **40B** + **104B** for the `Delta` object, in
+total, a minimum of **144B**.
+
+### `SkipList` memory layout
+
+Each object (`Vertex`, `Edge`) is placed inside a data structure
+called a `SkipList`. The `SkipList` has an additional overhead in terms of
+`SkipListNode` structure and `next_pointers`. Each `SkipListNode` has an
+additional **8B** element overhead and another **8B** for each of the `next_pointers`.
+
+It is impossible to know the exact size of the **#next_pointers** upfront, but
```

Comment:
this is what was bugging me in the notion file as well, but kate answered that this is common to use in math

Reply:
Ooo, now that you mentioned it, I guess this should represent like number of next_pointers. Didn't ring a bell at all while I was reading the docs, so not sure is this is the right way to annotate things for the docs .

Reply:
Yeah, I agree, we should write a number of, instead of # in the documentation. But, this sentence was not clear to me in the Notion too, left a comment [there](https://www.notion.so/memgraph/Scaling-Guide-c02176b4c5384fdda9b4ea4c8a322487#f5409c52daf74e10b25056899a65ee19).

Reply:
I will leave it as is for now, and we'll update upon getting more info